src/Implementation/State.cs

```diff
@@ -32,14 +32,16 @@ internal sealed class State : IState, IExecutable
             _system = system;
         }
 
-        public IEnumerable<IExecutable> Run()
+        public IEnumerable<IEnumerable<IExecutable>> Run()
         {
             _initialAction.Invoke(this);
+            yield return _forks;
```

Comment:
Let me know if this should be counted as an instruction or not.

Reply:
I'm not sure that I understand the question, but if I do, I don't think it should be counted. I think this initial action have come from some instruction already and to do so here would be double counting. I may be wrong though. Would this mean you would no longer have this `yield` here?

Reply:
Yeah, if this shouldn't be counted, because it was counted by the parent state, then it means we don't need to `yield` here.